src/birkhoff_polytope.jl

```diff
@@ -0,0 +1,589 @@
+            end
+            if x[row_orig, col_orig] <= eps()
+                if lmo.append_by_column
+                    d2[i, j] = missing
+                else
+                    d2[j, i] = missing
+                end
+            # the problem can only be integer types,
+            # either full-integer or mixed-integer.
+            elseif length(int_vars) == n^2 || idx_map_ub[orig_linear_idx] != 0
+                idx = length(int_vars) < n^2 ? idx_map_ub[orig_linear_idx] : orig_linear_idx
+                # interdict arc when fixed to zero
+                if ub[idx] <= eps()
+                    if lmo.append_by_column
+                        d2[i, j] = missing
+                    else
+                        d2[j, i] = missing
+                    end
+                else
+                    if lmo.append_by_column
+                        d2[i, j] = direction[row_orig, col_orig]
+                    else
+                        d2[j, i] = direction[col_orig, row_orig]
+                    end
+                end
+            else
+                if lmo.append_by_column
+                    d2[i, j] = direction[row_orig, col_orig]
+                else
+                    d2[j, i] = direction[col_orig, row_orig]
+                end
+            end
+        end
+    end
+
+    m = SparseArrays.spzeros(n, n)
+    for (i, j) in zip(fixed_to_one_rows, fixed_to_one_cols)
+        m[i, j] = 1
+    end
+
+    res_mat = Hungarian.munkres(d2)
+    (rows, cols, vals) = SparseArrays.findnz(res_mat)
+    @inbounds for i in eachindex(cols)
+        m[index_map_rows[rows[i]], index_map_cols[cols[i]]] = (vals[i] == 2)
+    end
+
+    return m
+end
+
+function Boscia.compute_inface_extreme_point(lmo::BirkhoffLMO, direction::AbstractVector{T}, x::AbstractVector{T}; kwargs...) where {T}
+    n = lmo.dim
+    direction = lmo.append_by_column ? reshape(direction, (n, n)) : transpose(reshape(direction, (n, n)))
+    x = lmo.append_by_column ? reshape(x, (n, n)) : transpose(reshape(x, (n, n)))
+    m = Boscia.compute_inface_extreme_point(lmo, direction, x; kwargs...)
+    m = if lmo.append_by_column
+            # Convert sparse matrix to sparse vector by columns
+            I, J, V = SparseArrays.findnz(m)
+            linear_indices = (J .- 1) .* n .+ I
+            SparseArrays.sparsevec(linear_indices, V, n^2)
+        else
+            # Convert sparse matrix to sparse vector by rows (transpose first)
+            mt = SparseArrays.sparse(LinearAlgebra.transpose(m))
+            I, J, V = SparseArrays.findnz(mt)
+            linear_indices = (J .- 1) .* n .+ I
+            SparseArrays.sparsevec(linear_indices, V, n^2)
+        end
+    
+    return m
+end
+
+"""
+LMO-like operation which computes a vertex minimizing in `direction` on the face defined by the current fixings.
+Fixings are maintained by the oracle (or deduced from `x` itself).
+"""
+function Boscia.dicg_maximum_step(lmo::BirkhoffLMO, direction, x; kwargs...)
```

Comment:
`function Boscia.dicg_maximum_step(lmo::BirkhoffLMO, direction, x; kwargs...)`

All the DICG functions should be FrankWolfe.function_name